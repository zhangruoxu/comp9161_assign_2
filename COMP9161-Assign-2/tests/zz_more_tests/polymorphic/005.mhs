main = 
  if True 
    then let g = letfun f x = x; in g
    else let g = letfun f x = x; in g
  ;
