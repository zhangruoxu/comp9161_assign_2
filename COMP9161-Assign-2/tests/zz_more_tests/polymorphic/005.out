main :: forall f. (f -> f) = if True
                                then let g :: forall a. (a -> a) = (letfun f :: (a -> a) x = x);
                                     in g
                                else let g :: forall d. (d -> d) = (letfun f :: (d -> d) x = x);
                                     in g;
